evalml/data_checks/default_data_checks.py

```diff
@@ -43,28 +51,46 @@ class DefaultDataChecks(DataChecks):
         DateTimeNaNDataCheck,
     ]
 
-    def __init__(self, problem_type, objective, n_splits=3):
+    def __init__(self, problem_type, objective, n_splits=3, datetime_column="index"):
+        default_checks = self._DEFAULT_DATA_CHECK_CLASSES
+        data_check_params = {}
```

Comment:
Changed the layout of this to make it cleaner and more easily understandable as to what data checks and params are being passed across different `problem_types`.